storm-core/src/jvm/backtype/storm/security/auth/authorizer/SimpleACLAuthorizer.java

```diff
@@ -50,6 +50,7 @@
     protected Set<String> _supervisors;
     protected IPrincipalToLocal _ptol;
     protected IGroupMappingServiceProvider _groupMappingProvider;
+    protected ImpersonationAuthorizer _impersonationAuthorizer;
```

Comment:
Ok, I am going to add this authorization as part of nimbus check-authorization! method. I will also add "nimbus.impersonation.authorizer" config to allow for overriding behavior. I will also add the logging changes there.